ui-v2/src/components/ui/tags-input.stories.tsx

```diff
@@ -0,0 +1,66 @@
+import { TagsInput } from "@/components/ui/tags-input.tsx";
+import { Meta, StoryObj } from "@storybook/react";
+import { expect, fn, userEvent, within } from "@storybook/test";
+import { ComponentProps, useState } from "react";
+
+export default {
+	title: "UI/TagsInput",
+	component: TagsInput,
+	args: {
+		value: [],
+		onChange: fn(),
+	},
+	argTypes: {
+		className: { control: { disable: true } },
+		type: { control: { disable: true } },
+	},
+	// To control input value in Stories via useState()
+	render: function Render(args: ComponentProps<typeof TagsInput>) {
+		const [value, setValue] = useState(args.value as string[]);
+		return (
+			<TagsInput
+				{...args}
+				value={value}
+				onChange={(x) => {
+					if (Array.isArray(x)) {
+						setValue(x);
+						args.onChange?.(x);
+					}
+				}}
+			/>
+		);
+	},
+} satisfies Meta<typeof TagsInput>;
+
+type Story = StoryObj<typeof TagsInput>;
+
+export const Empty: Story = {};
+
+export const DefaultValue: Story = {
+	args: { value: ["testTag", "testTag2"] },
+};
+
+export const TestAddRemove: Story = {
+	name: "Test: Add and remove",
+	play: async ({ canvasElement, step }) => {
+		const canvas = within(canvasElement);
+		const tagsInput: HTMLInputElement = canvas.getByRole("textbox");
+
+		await step("Add testTag", async () => {
+			await userEvent.type(tagsInput, "testTag");
+			await expect(tagsInput.value).toBe("testTag");
+			await userEvent.type(tagsInput, "{enter}");
+			await expect(tagsInput.value).toBe("");
+			await expect(canvas.queryByText("testTag")).not.toBeNull();
+		});
+
+		await step("Add testTag2", async () => {
+			await userEvent.type(tagsInput, "testTag2{enter}");
+		});
+
+		await step("Delete testTag", async () => {
+			await userEvent.click(canvas.getAllByRole("button")[0]);
+			await expect(canvas.queryByText("testTag")).toBeNull();
+		});
+	},
+};
```

Comment:
This is neat! I haven't seen tests written with Storybook before. Does this have any advantages over our current `vitest` + `testing-library` setup?

Reply:
I didn't set up the test runner for it on CI (I can add it to this PR or another PR if you're interested), so it is more for development than testing. 

- The play function further reduces the feedback loop on local-dev
- The time travel feature helps with debugging

Please see the [live version here](https://evanliu.dev/prefect/?path=/story/ui-tagsinput--test-add-remove).

<img width="725" alt="image" src="https://github.com/user-attachments/assets/8ccfa51d-cf63-4961-82bf-92e41b65da95">

Reply:
As the Storybook test also uses `vitest + testing-library`, here’s one potential workflow without a test runner for the Storybook tests:

1. Use the Storybook play function to facilitate local development.
2. Once development is completed, copy the Storybook test into the test suite.
3. Clean up the Storybook stories and tests, keep only the stories/tests for documentation purposes.